### PR TITLE
Test Onbuild Docker Images

### PR DESCRIPTION
**Pull request by kaxil**
We were previously not testing Onbuild images.

This PR:

- Use a single CircleCI step to test both base & onbuild images 
- Removes unused code

### CODE REVIEW

**Comment by kaxil** [comment]
@ashb  I have already tested this PR: https://app.circleci.com/pipelines/github/astronomer/ap-airflow/533/workflows/f6f021ae-9ac2-470c-adcc-0299a96dac00

**Comment by ashb** [comment]
Does it need to be two separate steps? Can't it be a single step in the pipeline, and a single`pytest` invocation that tests both images. (It's one build step, so one test step makes sense to me)

(This also makes the change smaller and easier to review)

**Comment by kaxil** [comment]
> Does it need to be two separate steps? Can't it be a single step in the pipeline, and a single`pytest` invocation that tests both images. (It's one build step, so one test step makes sense to me)
> 
> (This also makes the change smaller and easier to review)

Sure, will do

**Comment by kaxil** [comment]
@ashb Ready for review

File: .circleci/bin/test-airflow-image.py [hunk 1]
```diff
@@ -195,72 +207,38 @@ def scheduler(request):
 
 @pytest.fixture(scope='session')
 def docker_client(request):
-    """ This is a text fixture for the docker client,
-    should it be needed in a test
-    """
+    """ This is a text fixture for the docker client, should it be needed in a test """
```
Comment (ashb) [inline .circleci/bin/test-airflow-image.py:210]:
```suggestion
    """ This is a test fixture for the docker client, should it be needed in a test """
```

File: .circleci/config.yml.j2 [hunk 2]
```diff
@@ -111,9 +110,8 @@ workflows:
             - build-{{ airflow_version }}-{{ distribution }}
       - test:
           name: test-{{ airflow_version }}-{{ distribution }}-onbuild
```
Comment (ashb) [inline .circleci/config.yml.j2:112]:
```suggestion
          name: test-{{ airflow_version }}-{{ distribution }}-images
```

Comment (kaxil) [reply .circleci/config.yml.j2]: Fixed

**Pull request merged**